src/reducer.js

```diff
@@ -30,6 +31,15 @@ const reducers = {
     };
   },
 
+  setRestaurant(state, { payload: { restaurant } }) {
+    return {
+      ...state,
+      restaurant: {
+        ...restaurant,
+      },
+    };
+  },
```

Comment:
```suggestion
  setRestaurant(state, { payload: { restaurant } }) {
    return {
      ...state,
      restaurant
    };
  },
```

이렇게 처리하는 것과 같지 않나요? 🤔

Reply:
ㅎㅎㅎ 이전에 initialState의 restaurant를 key 를 일일히 다 정의하고 key 하나하나당 일일히 value 값을 주어봤었습니다, (deep merge 고려 such as menuItems, reviews.... 어차피 새로운 배열로 replace 하는 것인데... 괜한 고려를...) 

다시 원상태로 복구할 때 신경을 못썼네요. 🔨🔨🔨🔨🔨🔨🔨🔨🔨🔨🔨🔨

Reply:
저는 타입스크립트를 주로 사용하기 때문에 타입 인터페이스를 정의해주고 항상 초기화를 함께 해줍니다 😊

초기 상태값이 복잡해진다면 객체를 쪼개는 것도 가독성에 좋을 것 같네요 :)

```javascript
//예시
const initialRestaurant = {
    id: null,
    name: '',
    address: '',
    menuItems: [],
    reviews: [],
    information: '',
  };

const initialState = {
  regions: [],
  categories: [],
  restaurants: [],
  selectedRegion: null,
  selectedCategory: null,
  restaurant: initialRestaurant,
};
```